### PR TITLE
[rebalancing] cache: add cache functions for tagging containers.

### PR DESCRIPTION
**Pull request by klihub**
Add functions for manipulating container tags, which are basically dynamic labels which exist only in the scope of the resource manager itself. We'll use tags to attach dynamically changing attributes to running containers based on the collected metrics data.

### CODE REVIEW

File: pkg/cri/resource-manager/cache/cache.go [hunk 1]
```diff
@@ -344,6 +351,7 @@ type container struct {
 	Mounts        map[string]*Mount   // mounts
 	Devices       map[string]*Device  // devices
 	TopologyHints sysfs.TopologyHints // Set of topology hints for all containers within Pod
+	Tags          map[string]string   // container tags (local dyanmic labels)
```
Comment (marquiz) [inline pkg/cri/resource-manager/cache/cache.go]: s/dyanmic/dynamic/

File: pkg/cri/resource-manager/cache/container.go [hunk 2]
```diff
@@ -790,3 +793,20 @@ func (c *container) HasPending(controller string) bool {
...
+}
+
+func (c *container) SetTag(key string, value string) (string, bool) {
+	prev, ok := c.Tags[key]
+	c.Tags[key] = value
+	return prev, ok
+}
+
+func (c *container) DeleteTag(key string) (string, bool) {
+	prev, ok := c.Tags[key]
```
Comment (marquiz) [inline pkg/cri/resource-manager/cache/container.go]: nit: `prev` is a bit strange name in this context :smile:

Comment (klihub) [reply pkg/cri/resource-manager/cache/container.go]: True, I'll change it to `value`...

File: pkg/cri/resource-manager/cache/cache.go (hunk 1 above)
Comment (klihub) [reply pkg/cri/resource-manager/cache/cache.go]: Thanks, fixed.

**Pull request merged**